nova_vm/src/ecmascript/builtins/text_processing/string_objects/string_prototype.rs

```diff
@@ -645,8 +645,85 @@ impl StringPrototype {
         ))
     }
 
-    fn replace(_agent: &mut Agent, _this_value: Value, _: ArgumentsList) -> JsResult<Value> {
-        todo!()
+    /// ### [22.1.3.19 String.prototype.replace ( searchValue, replaceValue )](https://tc39.es/ecma262/multipage/text-processing.html#sec-string.prototype.replace)
+    fn replace(agent: &mut Agent, this_value: Value, args: ArgumentsList) -> JsResult<Value> {
+        // 1. Let O be ? RequireObjectCoercible(this value).
+        let o = require_object_coercible(agent, this_value)?;
+
+        let search_value = args.get(0);
+        let replace_value = args.get(1);
+
+        // 2. If searchValue is neither undefined nor null, then
+        if !search_value.is_null() && !search_value.is_undefined() {
+            let symbol = WellKnownSymbolIndexes::Replace.into();
+            // If replacer is not undefined, Return ? Call(replacer, searchValue, « O, replaceValue »).
```

Comment:
**issue:** We're missing the comment that explains how `replacer` is to be found. (I'm quite interested in that because I hope to god that we don't actually have to look up the replacer function for strings as well :cry: )

Reply:
I've added the comment with the instructions from the spec. The replacer is found with the `get_method` function.